File: hash-tables/twoSum.js

const array=[2,7,11,15];

function twoSum(array, target) {
    let list = {};
    for(let i=0;i<=array.length-1;i++){
      let item = target - array[i];
      if(!list[item]){
        list[item] = true
      }
      if(array[i] in list){
        console.log("founded", array[i], "plus",target-array[i])
      }
    }

};
twoSum(array,9);

File: hash-tables/twoSum.py
arr = [2,7,11,15]

def twoSum(arr, number):
    table = {}
    for i in range(len(arr)):
        target = number - arr[i]
        if(arr[i] not in table):
            table[arr[i]] = target
        #print(table)
        #print(table[arr[i]])
        if(table[arr[i]] in arr):
            print(arr[i])


twoSum(arr,18)

